uniprotkb-rest/src/test/java/org/uniprot/api/uniprotkb/controller/UniProtKBSearchControllerIT.java

```diff
@@ -727,17 +728,37 @@ void badDefaultSearchWithMatchedFields() throws Exception {
                 getMockMvc()
                         .perform(
                                 get(SEARCH_RESOURCE
-                                                + "?query=gene:Fibroblast&fields=accession,gene_primary&showMatchedFields=true")
+                                                + "?query=gene:FGFR2&fields=accession,gene_primary&showSingleTermMatchedFields=true")
                                         .header(ACCEPT, APPLICATION_JSON_VALUE));
 
         // then
-        response.andDo(log())
-                .andExpect(status().is(HttpStatus.BAD_REQUEST.value()))
-                .andExpect(
-                        content()
-                                .string(
-                                        containsString(
-                                                "Term information will only be returned for single value searches that do not specify a field")));
+        response.andDo(print())
+                .andExpect(status().is(HttpStatus.OK.value()))
+                .andExpect(header().string(HttpHeaders.CONTENT_TYPE, APPLICATION_JSON_VALUE))
+                .andExpect(jsonPath("$.results.*.primaryAccession", contains("P21802")))
+                .andExpect(jsonPath("$.matchedFields").doesNotExist());
+    }
+
+    @Test
+    void defaultPhrasalSearchWithMatchedFieldsDoesNotReturnMatchedFields() throws Exception {
```

Comment:
defaultPhrasalSearchWithMatchedFieldsDoesNotReturnMatchedFields -> multiwordDefaultSearchWithMatchedFieldsDoesNotReturnMatchedFields

(this isn't a phrase query, because no quotes)